opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -342,11 +342,17 @@ namespace Opm {
                     setRepRadiusPerfLength();
                 }
             }
+        } catch (std::runtime_error& e) {
+            exc_type = ExceptionType::RUNTIME_ERROR;
+        } catch (std::invalid_argument& e) {
+            exc_type = ExceptionType::INVALID_ARGUMENT;
+        } catch (std::logic_error& e) {
+            exc_type = ExceptionType::LOGIC_ERROR;
```

Comment:
We should typically catch objects as "reference-to-const", not as "reference-to-mutable".